### PR TITLE
ci: add cluster upgrade test

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Adds cluster upgrade test as part of CI nightly runs for AKS

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/174

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by aramase** [comment]
Successful test run with upgrade: https://dev.azure.com/azure/secrets-store-csi-driver-provider-azure/_build/results?buildId=20928&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=04b98c25-2b2b-580a-44f7-07920a0b2eeb

**Comment by aramase** [comment]
Successful test run: https://dev.azure.com/azure/secrets-store-csi-driver-provider-azure/_build/results?buildId=20935&view=logs&j=53fa2608-c467-5776-6d15-db4adfc07c83

File: .pipelines/templates/aks-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      AKS_UPGRADE_VERSION=$(az aks get-upgrades -g ${AZURE_CLUSTER_NAME} -n ${AZURE_CLUSTER_NAME} --query 'max(controlPlaneProfile.upgrades[].kubernetesVersion)' -otsv)
+      echo "##vso[task.setvariable variable=AKS_UPGRADE_VERSION]${AKS_UPGRADE_VERSION}"
+    displayName: "Get upgrade version"
+
+  - script: |
+      echo "Upgrading to Kubernetes ${AKS_UPGRADE_VERSION}"
+      az aks upgrade -g ${AZURE_CLUSTER_NAME} -n ${AZURE_CLUSTER_NAME} -k ${AKS_UPGRADE_VERSION} --yes
+
+      # Sleep for 120 seconds to wait for nodes and pods to become ready
+      sleep 2m
```
Comment (ritazh) [inline .pipelines/templates/aks-upgrade.yaml:12]: is 2m enough?

Comment (aramase) [reply .pipelines/templates/aks-upgrade.yaml:12]: The 2m sleep is after upgrade to check health of the components. `az aks upgrade` is blocking until the upgrade is complete.

**Review by ritazh — APPROVED** [review]

**Pull request merged**